packages/web-features-figuring/src/baseline/index.ts

```diff
@@ -0,0 +1,244 @@
+
+  let baseline_low_date = null;
+  if (baseline !== false) {
+    baseline_low_date = latestDate(
+      statuses.map((status) => status.baseline_low_date),
+    );
+  }
+
+  let baseline_high_date = null;
+  if (baseline === "high") {
+    baseline_high_date = latestDate(
+      statuses.map((status) => status.baseline_high_date),
+    );
+  }
+
+  return {
+    baseline,
+    baseline_low_date,
+    baseline_high_date,
+    support: minSupport(statuses.map((status) => status.support)),
+    toJSON: function () {
+      return jsonify(this);
+    },
+  };
+}
+
+function calculate(compatKey: string, compat: Compat): SupportStatus {
+  const f = feature(compatKey);
+  const s = support(f, browsers(compat), compat);
+
+  const allReleases = f.supportedBy({ only: browsers(compat), compat });
+  const isBaselineLow = lowReleases(compat).every((r) =>
+    allReleases.includes(r),
+  );
+  const isBaselineHigh = highReleases(compat).every((r) =>
+    allReleases.includes(r),
+  );
+
+  const baseline =
+    (isBaselineHigh ? "high" : false) || (isBaselineLow ? "low" : false);
+
+  let baseline_low_date = null;
+  let baseline_high_date = null;
+  if (isBaselineLow) {
+    const initialReleases = [...s.entries()].map(([, r]) => r);
+    const keystoneRelease = initialReleases
+      .sort((a, b) => {
+        assert(a !== undefined);
+        assert(b !== undefined);
+        return Temporal.PlainDate.compare(
+          a.date() ?? VERY_FAR_FUTURE_DATE,
+          b.date() ?? VERY_FAR_FUTURE_DATE,
+        );
+      })
+      .pop();
+
+    assert(keystoneRelease instanceof Release);
+    baseline_low_date = (keystoneRelease.date() ?? VERY_FAR_FUTURE_DATE)
+      .toString()
+      .slice(0, 10);
+
+    if (isBaselineHigh) {
+      const date = keystoneRelease.date();
+      assert(date !== null);
+      baseline_high_date = date
+        .add(BASELINE_LOW_TO_HIGH_DURATION)
+        .toString()
+        .slice(0, 10);
+    }
+  }
+
+  return {
+    compatKey,
+    baseline,
+    baseline_low_date,
+    baseline_high_date,
+    support: s,
+    toJSON: function () {
+      return jsonify(this);
+    },
+  };
+}
+
+function withAncestors(compatKey: string, compat: Compat): string[] {
+  const items = compatKey.split(".");
+  const ancestors: string[] = [];
+
+  let current = items.shift();
+  while (items.length) {
+    current = `${current}.${items.shift()}`;
+
+    const data = compat.query(current);
+    if (typeof data === "object" && data !== null && "__compat" in data) {
+      ancestors.push(current);
+    }
+  }
+  return ancestors;
+}
+
+function minStatus(statuses: SupportStatus[]): "high" | "low" | false {
+  let bestOf: "high" | "low" = "high";
+  for (const { baseline } of statuses) {
+    if (baseline === false) {
+      return false;
+    }
+    if (baseline === "low") {
+      bestOf = "low";
+    }
+    if (baseline === "high" && bestOf === "high") {
+      bestOf = "high";
+    }
+  }
+  return bestOf;
+}
+
+function minSupport(
```

Comment:
Can you document the intended behavior of this helper? It looks like it will find the earliest supported version even in ship-remove-ship situations. Some code might need that, but not when computing Baseline, right?